.dotnet/src/Custom/Chat/OpenAIChatModelFactory.cs

```diff
@@ -0,0 +1,175 @@
+        ChatFinishReason? finishReason = null,
+        DateTimeOffset createdAt = default,
+        string model = null,
+        string systemFingerprint = null,
+        ChatTokenUsage usage = null)
+    {
+        contentUpdate ??= new List<ChatMessageContentPart>();
+        toolCallUpdates ??= new List<StreamingChatToolCallUpdate>();
+        contentTokenLogProbabilities ??= new List<ChatTokenLogProbabilityInfo>();
+
+        InternalChatCompletionStreamResponseDelta delta = new InternalChatCompletionStreamResponseDelta(
+            contentUpdate.ToList(),
+            functionCallUpdate,
+            toolCallUpdates.ToList(),
+            role,
+            serializedAdditionalRawData: null);
+
+        InternalCreateChatCompletionStreamResponseChoiceLogprobs logprobs = new InternalCreateChatCompletionStreamResponseChoiceLogprobs(
+            contentTokenLogProbabilities.ToList(),
+            serializedAdditionalRawData: null);
+
+        IReadOnlyList<InternalCreateChatCompletionStreamResponseChoice> choices = [
+            new InternalCreateChatCompletionStreamResponseChoice(
+                delta,
+                logprobs,
+                finishReason,
+                index: 0,
+                serializedAdditionalRawData: null)
+        ];
+
+        return new StreamingChatCompletionUpdate(
+            id,
+            choices,
+            createdAt,
+            model,
+            systemFingerprint,
+            InternalCreateChatCompletionStreamResponseObject.ChatCompletionChunk,
+            usage,
+            serializedAdditionalRawData: null);
+    }
+
+    /// <summary> Initializes a new instance of <see cref="OpenAI.Chat.StreamingChatFunctionCallUpdate"/>. </summary>
+    /// <returns> A new <see cref="OpenAI.Chat.StreamingChatFunctionCallUpdate"/> instance for mocking. </returns>
+    public static StreamingChatFunctionCallUpdate StreamingChatFunctionCallUpdate(string functionArgumentsUpdate = null, string functionName = null)
+    {
+        return new StreamingChatFunctionCallUpdate(
+            functionArgumentsUpdate,
+            functionName,
+            serializedAdditionalRawData: null);
+    }
+
+    /// <summary> Initializes a new instance of <see cref="OpenAI.Chat.StreamingChatToolCallUpdate"/>. </summary>
+    /// <returns> A new <see cref="OpenAI.Chat.StreamingChatToolCallUpdate"/> instance for mocking. </returns>
+    public static StreamingChatToolCallUpdate StreamingChatToolCallUpdate(int index = default, string id = null, ChatToolCallKind kind = default, string functionName = null, string functionArgumentsUpdate = null)
+    {
+        InternalChatCompletionMessageToolCallChunkFunction function = new InternalChatCompletionMessageToolCallChunkFunction(
+            functionName,
+            functionArgumentsUpdate,
+            serializedAdditionalRawData: null);
+
+        return new StreamingChatToolCallUpdate(
+            index,
+            id,
+            kind,
```

Comment:
I'm allowing customers to set `kind` here but at the moment we only support the value `Function`. Another possibility would be removing this parameter altogether and forcibly setting `Kind` to `Function`.

Reply:
This is fine!